### DISCUSSION TITLE
stiker full

### DISCUSSION THREAD

**Discussion by techwiz37** [tulir/whatsmeow#483] [General]
how to make a full sticker without cropping

**Comment by ghost** [comment]
Sorry you mean without cropping or without distorted ratio aspects like this?
![gambar](https://github.com/tulir/whatsmeow/assets/143764626/984d78fb-f325-4553-873e-a4b6222dc92d)

If your intention is distorted like that you need to adjust the image that you will send as a sticker. WhatsApp stickers have aspects of 1: 1 ratio and WhatsApp application forces images that are sent as setickers following that ratio, so if you use images with ratios other than 1: 1, the WhatsApp application will stretch the image to have the appropriate ratio, as a result of it. what we send will be distorted as the example I gave above.
In cases like this I use the following FFMPEG command to set the image that I will send to meet the criteria earlier.
```bash
ffmpeg -i image.jpg -vcodec libwebp -vf "scale='min(320,iw)':'min(320,ih)':force_original_aspect_ratio=decrease,fps=23,pad=320:320:-1:-1:color=white@0.0,split[a][b];[a]palettegen=reserve_transparent=on:transparency_color=ffffff[p];[b][p]paletteuse" -f webp output.webp
```
So here the way it works is FFMPEG will make a transparent image with a size of 320x320 and scatter input images to have a height or maximum width of 320px and then combine it with the transparent image earlier so as to produce images with aspects of 1: 1 ratio and transparent background. So the WhatsApp application does not stretch the image that we make a sticker.
but the shortcomings of that method if our images are smaller than 320x320 then the image will have transparent gaps around our images and our images will look very small.

**Comment by techwiz37** [comment]
solved tq
